sql/analyzer/tables.go

```diff
@@ -98,28 +98,28 @@ func getResolvedTable(node sql.Node) *plan.ResolvedTable {
 }
 
 // getTablesByName takes a node and returns all found resolved tables in a map.
+// This function will not look inside sql.OpaqueNodes (like plan.SubqueryAlias).
 func getTablesByName(node sql.Node) map[string]*plan.ResolvedTable {
 	ret := make(map[string]*plan.ResolvedTable)
-
-	transform.Inspect(node, func(node sql.Node) bool {
+	// TODO: We should change transform.Inspect to not walk the children of sql.OpaqueNodes (like transform.Node)
+	//  and add a transform.InspectWithOpaque that does.
+	//  Using transform.Node here achieves the same result without a large refactor.
```

Comment:
Great observation/TODO. There are several small inconsistencies in the `transform.*` functions that would be great to tidy up one day.